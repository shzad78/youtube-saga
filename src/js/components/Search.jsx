import React from 'react';
import PropTypes from 'prop-types';

export default class Search extends React.Component {
  constructor() {
    super();

    this.state = {
      value: ''
    };

    this.localHandler = this.localHandler.bind(this);
  }

  localHandler(e) {
    const { getVideos } = this.props;

    this.setState({ value: e.target.value });
    getVideos(e.target.value);
  }

  render() {
    return (
      <div className="search col-md-6 col-md-offset-4 form-inline">
        <input
          type="text"
          className="form-control"
          onChange={this.localHandler}
          value={this.state.value}
        />
        <button className="btn">
          <span className="glyphicon glyphicon-search " />
        </button>
      </div>
    );
  }
}

Search.propTypes = {
  getVideos: PropTypes.func.isRequired
};
