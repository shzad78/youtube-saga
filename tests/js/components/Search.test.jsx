import React from 'react';
import Search from '../../../src/js/components/Search';

//Below are related to enzyme
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });


describe('Testing Search Component',()=>{
  const getVideos = jest.fn();
  const wrapper = shallow(<Search getVideos={getVideos} />);
  
  it('renders the Search Component', ()=>{  
    expect(wrapper).toMatchSnapshot();
  });
  
  it('it runs the getVideos function with correct args', ()=>{
    wrapper.find('input').simulate('change', {target:{value:'java'}})
    expect(getVideos.mock.calls[0]).toEqual(['java']);
  });

});

