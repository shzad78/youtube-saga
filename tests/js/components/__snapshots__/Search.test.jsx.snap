// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Search Component renders the Search Component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    getVideos={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="form-control"
          onChange={[Function]}
          type="text"
          value=""
        />,
        <button
          className="btn"
        >
          <span
            className="glyphicon glyphicon-search "
          />
        </button>,
      ],
      "className": "search col-md-6 col-md-offset-4 form-inline",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control",
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="glyphicon glyphicon-search "
          />,
          "className": "btn",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "glyphicon glyphicon-search ",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="form-control"
            onChange={[Function]}
            type="text"
            value=""
          />,
          <button
            className="btn"
          >
            <span
              className="glyphicon glyphicon-search "
            />
          </button>,
        ],
        "className": "search col-md-6 col-md-offset-4 form-inline",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="glyphicon glyphicon-search "
            />,
            "className": "btn",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "glyphicon glyphicon-search ",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
